cristin-project/src/main/java/no/unit/nva/cristin/projects/model/cristin/CristinFundingSource.java

```diff
@@ -39,6 +45,20 @@ public void setFundingSourceName(Map<String, String> fundingSourceName) {
         this.fundingSourceName = fundingSourceName;
     }
 
+    /**
+     * Extracts funding source code from URI giving a valid Cristin source code.
+     */
+    public static String extractFundingSourceCode(URI source) {
+        if (isNull(source)) {
+            return null;
+        }
+        var sourceAsText = source.toString();
+        var lastElementIndexStart = sourceAsText.lastIndexOf(PATH_DELIMITER) + 1;
+        var rawSourceCode = sourceAsText.substring(lastElementIndexStart);
```

Comment:
I think there is a method in commons for this UriWrapper::getLastPathElement, but I wonder if a trailing slash breaks this…